stumpy/core.py

```diff
@@ -2177,7 +2181,7 @@ def preprocess_non_normalized(T, m):
     return T, T_subseq_isfinite
 
 
-def preprocess_diagonal(T, m, T_subseq_isconstant=None):
+def preprocess_diagonal(T, m, copy=True, T_subseq_isconstant=None):
```

Comment:
Be very careful. When you insert a keyword argument in the middle of a function, it may/can cause conflicts. You must go and ensure that all calls to `preprocess_diagonal` have `T_subseq_isconstant` explicitly defined by keyword and NOT by position. I just verified that this was `True` but I wanted to call it out for the future. Otherwise, if there were some code that was currently doing:

```
core.preprocess_diagonal(T, m, None)
```

Then, we could end up with `copy=None`, which would lead to unexpected behavior. As ugly/inconsistent as it may look, it is best to append `copy=True` to the end of the list of parameters. In this case, it is probably "okay" but "power users" may be calling `preprocess_diagonal` themselves for other purposes and it would cause problems

Reply:
> Then, we could end up with copy=None, which would lead to unexpected behavior. As ugly/inconsistent as it may look, it is best to append copy=True to the end of the list of parameters

> In this case, it is probably "okay" but "power users" may be calling preprocess_diagonal themselves for other purposes and it would cause problems

Right! I thought if the tests are passing, then we should be okay. However, I did not think about that particular scenario. Got it! I will revise the code accordingly. 

> Be very careful.

> but I wanted to call it out for the future. 

Thanks for bring it to my attention! 👍